### PR TITLE
test: block and memblock serialization

### PR DESCRIPTION
**Pull request by pinheadmz**
Cleans up `block-test.js` and ports the test from https://github.com/handshake-org/hsd/pull/249

### CODE REVIEW

File: test/block-test.js [hunk 1]
```diff
@@ -4,19 +4,43 @@
...
+        bits: 0,
+        nonce: 0,
+        extraNonce: Buffer.alloc(24, 5),
+        mask: Buffer.alloc(32, 6)
+      });
+      assert(block);
+    });
+
+    it('should deserialize and reserialze block', async () => {
+      raw = block.toRaw();
```
Comment (tynes) [inline test/block-test.js]: Nit: use `encode`/`decode` over `toRaw` and `fromRaw`

Comment (pinheadmz) [reply test/block-test.js]: thanks updated

**Comment by pinheadmz** [comment]
Notice my dummy block has 0 transactions (not even a coinbase), looks like `Memblock` doesn't parse those anyway, and leaving them out makes this a more acute unit test (not having to include the `TX` module...) Does that seem ok?

**Comment by pinheadmz** [comment]
Added a TX to the block and switched to `fromJSON()`.

File: test/block-test.js [hunk 2]
```diff
@@ -4,19 +4,72 @@
...
+            outputs: [
+              {
+                value: 1234567,
+                address: 'rs1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqn6kda'
+              }
+            ]
+          }
+        ]
+      });
+      assert(block);
```
Comment (boymanjor) [inline test/block-test.js]: How do you feel about asserting the individual attributes on the block primitive?

Comment (pinheadmz) [reply test/block-test.js]: Good idea, done at 3419d7cfece14db95cc899f996a351c8527f8387

**Pull request merged**